src/ScreenHeader/HeaderButton.tsx

```diff
@@ -1,21 +1,75 @@
 import {AccessibilityProps, TouchableOpacity} from 'react-native';
 import React from 'react';
 import insets from '../utils/insets';
+import {LeftButtonProps, RightButtonProps} from './index';
```

Comment:
Unsure of the proper way to handle types here. Now we have a circular dependency (which works, but is kind of iffy) and it is kind of odd with proper ownership. But I also see the challenge of right vs left button. But my suggestion is to have there be a `HeaderButtonProps` like so in `HeaderButton.tsx`

```
export type ButtonModes = 'back' | 'cancel' | 'close' | 'home' | 'chat'l;
export type HeaderButtonProps = {
  type: ButtonModes;
  onPress?: () => void;
} & AccessibilityProps;
```

and in `ScreenHeader/index.tsx`

```
export type LeftButtonProps = HeaderButtonProps & {
  type: Exclude<ButtonModes, 'chat'>;
};

export type RightButtonProps = Omit<HeaderButtonProps, 'onPress'> & {
  type: 'chat';
};
```

Not perfect, but I think maybe it's better